db/hints/manager.cc

```diff
@@ -778,11 +778,8 @@ bool manager::end_point_hints_manager::sender::send_one_file(const sstring& fnam
 
     // update the next iteration replay position if needed
     if (ctx_ptr->state.contains(send_state::segment_replay_failed)) {
-        if (ctx_ptr->state.contains(send_state::restart_segment)) {
```

Comment:
Why is it safe to remove this case?

Reply:
We used to set this flag only when inserting to rps_set failed. Now, this flag is never set, so we can remove this case.